### PR TITLE
Qa 75 production gitlab pipeline

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    # fail if any have been modified.
+    - if [ -n "$(go fmt)" ]; then echo 'Code is not formatted with "go fmt"'; false; fi
+    # Perform static code analysys
+    - go vet `go list ./... | grep -v vendor`
+    # Fail builds when the cyclomatic complexity reaches 15 or more
+    - gocyclo -over 15 `find . -iname '*.go' | grep -v 'vendor' | grep -v '_test.go'`
+    # Verify that the Swagger docs are valid
+    - swagger-cli validate docs/*.yml
+    # Verify that the Swagger docs follow the autodeployment requirements
+    - python2.7 verify_docs.py `find docs -name "*.yml"`
```
Comment (lluiscampos) [inline .gitlab-ci.yml]: We can change this to Python 3 call once we merge [this PR](https://github.com/mendersoftware/autodocs/pull/1).

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  script:
+    - apk add git bash
+    - 'export SHARED_PATH="$(dirname ${CI_PROJECT_DIR})"'
+    - git clone https://github.com/mendersoftware/integration.git ${SHARED_PATH}/mender-integration
+    - mkdir -p ${SHARED_PATH}/tests
+    - mv ${SHARED_PATH}/mender-integration/extra/travis-testing/* docs/* deployments mender-artifact tests/* ${SHARED_PATH}/tests
+    - docker load -i acceptance_testing_image.tar
+    - docker load -i tests_image.tar
+    - TESTS_DIR=${SHARED_PATH}/tests ${SHARED_PATH}/tests/run-test-environment acceptance ${SHARED_PATH}/mender-integration ${SHARED_PATH}/tests/docker-compose-acceptance.yml ;
+    - TESTS_DIR=${SHARED_PATH}/tests ${SHARED_PATH}/tests/run-test-environment acceptance ${SHARED_PATH}/mender-integration ${SHARED_PATH}/tests/docker-compose-acceptance.mt.yml ;
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply .gitlab-ci.yml]: My bad understanding here. Looks good

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    - mv ${SHARED_PATH}/tests/coverage-acceptance.txt $CI_PROJECT_DIR
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply .gitlab-ci.yml]: And here :)

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  before_script: []
+  services:
+    - docker:dind
+  stage: test
+  dependencies:
+    - test:build_acceptance
+    - test:prepare_acceptance
+  script:
+    - apk add git bash
+    - 'export SHARED_PATH="$(dirname ${CI_PROJECT_DIR})"'
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply .gitlab-ci.yml]: :+1:

File: .gitlab-ci.yml (hunk 1 above)
Comment (lluiscampos) [reply .gitlab-ci.yml]: The mentioned PR is now merged, can we please switch to Python 3? Also in `device_auth`, `deviceadm`, `inventory`, `tenantadm`, `mender-cli` and `useradm`.

File: .gitlab-ci.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mzedel) [reply .gitlab-ci.yml]: btw. that was adapted from a Gitlab suggestion and seemed to work, I guess if the whole line is just a literal the literals content will be executed instead 🤷‍♂

**Comment by mzedel** [comment]
Good point to clean all this. I changed the ones in question and removed the part from `mender-cli` - turns out it's not used there at all 🎉 and since it's deprecated I won't change `deviceadm`.

**Comment by lluiscampos** [comment]
Re-running integ tests just to be in the safe side. See [Jenkins build](https://mender-jenkins.mender.io/job/mender-builder/3877/).

**Pull request merged**